### ISSUE TITLE
White screen appears during scene preview startup 

### ISSUE THREAD

**Issue by RoryMcMaster** [godotengine/godot#13381]
**Operating system or device, Godot version, GPU Model and driver (if graphics related):**
3.0, GTX 950 - Driver: 388.31

**Issue description:**
Recently upgraded the PC (from AMD 7850) to a GTX 950.

However, whenever scene previewing there is a white screen that appears, and it stays there for roughly 1-2 seconds. Not sure if giving a project example would be ideal since it happens any project. But a gif video might help:

![](https://i.gyazo.com/786c290804b13728f64e234eb1ae4a74.gif)
 
Sorry if this seems like a first world problem, but we're finally converting a Unity game to 3.0 and don't remember  this on the 7850.

**Comment by RoryMcMaster** [comment]
Also, forgot to add. On exported games, this the white screen appears but it's instant (no wait time)

**Comment by Krakean** [comment]
@RoryMcMaster fixed in PR #13383

**Comment by RoryMcMaster** [comment]
@Krakean Wow, that was quick.

Just re-compiled.

Unfortunately, the delay is still there:
![](https://i.gyazo.com/234d9b9644bce9225a2e1abea707e37d.gif)

However, it's a lot smoother now since the white screen is gone. So your PR is actually still really good.

This issue might be deeper. Doing some searching and finally stumbled upon:
https://discourse.libsdl.org/t/massive-startup-time-difference-between-opengl-window-and-non-opengl-window/22824/11

Which they linked this below:
https://twitter.com/nothings/status/891335303130005504

Which is:
![](https://pbs.twimg.com/media/DF6og0-VwAA4MeA.jpg)

Looks like an inherently driver issue of initializing the start up of OpenGL's `ChoosePixelFormat`

Am going to install a different driver version and report back.

**Comment by Krakean** [comment]
@RoryMcMaster 
>Looks like an inherent driver issue of initializing the start up of OpenGL'
>Edit: Tried the 378.66 version. Delay still evident unfortunately. Damn. Must dig deeper.

Nah, it's OpenGL. Just deal with it.
Some day Godot will nice to have DirectX11-based renderer implementation just for Windows users.

**Comment by RoryMcMaster** [comment]
@Krakean Yeah, thank you.  Just curious, is there is any reason why an exported game doesn't have the start up delay, when scene previewing does? Is it possible as a setting to keep OpenGL initialized  in the background while editing?

**Comment by akien-mga** [comment]
> is there is any reason why an exported game doesn't have the start up delay, when previewing in the editor does?

Exported game: optimized release binary.
Preview in the editor: debug build initializing the connection to the Godot debugger, etc. Some delay for debug builds is normal.

**Comment by RoryMcMaster** [comment]
Yeah very true. Just seems odd with the 7850 the delay was never there in the editor, or on compiled games. But now w/ GTX 950, the delay isn't there on exported games, but is with the editor.

**Comment by akien-mga** [comment]
Maybe Nvidia's drivers provide better debugging info than AMD's. At any rate, 1-2 s of delay is totally normal and is due to the combination of many factors.

**Comment by RoryMcMaster** [comment]
@akien-mga Here is something very interesting

If you close the editor, and then run this command:

```
C:\Users\Nick>"C:\Users\Rory\Desktop\GodotEditor.exe" "--path" "C:/Users/Rory/Do
cuments/1234" "--remote-debug" "127.0.0.1:6007" "--allow_focus_steal_pid" "3000"
 "--position" "448,240"
```

Game will open instantly, just like on the 7850. So my OpenGL initialization thoughts are now void.

**Comment by akien-mga** [comment]
That may be due to the initialization of the new remote scene debugger in the editor, which probably hadn't been merged previously while you were using the 7850.

**Comment by RoryMcMaster** [comment]
Yeah, good idea. Transferred the new compiled (today's) executable to the 7850 computer and tested it. The delay is non-existent anywhere.

What is concluded to help isolate  

- GTX 950 is instant on scene previewing when running via CMD `GodotEditor.exe "--path" "C:/Users/Rory/Do
cuments/1234" "--remote-debug" "127.0.0.1:6007" "--allow_focus_steal_pid" "3000"
 "--position" "448,240"`

- GTX 950 **is not** _instant when running that command_, **if you have** the GodotEditor.exe open, or, **by clicking run** in editor directly.


When Godot's editor is open, something is happening with it, and the GTX 950. Not sure how to further help, unfortunately :(

**Comment by Zylann** [comment]
Sounds similar to https://github.com/godotengine/godot/issues/11939
